sdk/storage/storage-blob/samples/typescript/advanced.ts

```diff
@@ -100,13 +118,33 @@ async function main() {
   // downloadToBuffer is only available in Node.js
   const fileSize = fs.statSync(localFilePath).size;
   const buffer = Buffer.alloc(fileSize);
-  await blockBlobClient.downloadToBuffer(buffer, 0, undefined, {
-    abortSignal: AbortController.timeout(30 * 60 * 1000), // Abort uploading with timeout in 30mins
-    blockSize: 4 * 1024 * 1024, // 4MB block size
-    concurrency: 20, // 20 concurrency
-    onProgress: (ev) => console.log(ev)
-  });
-  console.log("downloadToBuffer success");
+  try {
+    await blockBlobClient.downloadToBuffer(buffer, 0, undefined, {
+      abortSignal: AbortController.timeout(30 * 60 * 1000), // Abort uploading with timeout in 30mins
+      blockSize: 4 * 1024 * 1024, // 4MB block size
+      concurrency: 20, // 20 concurrency
+      onProgress: (ev) => console.log(ev)
+    });
+    console.log("downloadToBuffer succeeds");
+  } catch (err) {
+    console.log(
+      `downloadToBuffer failed, requestId - ${err.details.requestId}, statusCode - ${err.statusCode}, errorCode - ${err.details.errorCode}`
+    );
+  }
+
+  // Archive the blob - Log the error codes
+  await blockBlobClient.setAccessTier("Archive");
+  try {
+    // Downloading an archived blockBlob fails
+    console.log("// Downloading an archived blockBlob fails...");
+    await blockBlobClient.download();
+  } catch (err) {
+    // BlobArchived	Conflict (409)	This operation is not permitted on an archived blob.
+    console.log(
+      `requestId - ${err.details.requestId}, statusCode - ${err.statusCode}, errorCode - ${err.details.errorCode}`
```

Comment:
AbortError doesn't have `details`, `err.statusCode` and `err.details.errorCode`.

So we should not make customer assume error `err` they caught can have these properties. Can we update the documentation?

Reply:
Documentation has to be updated in the abort-controller package I guess.

cc - @chradek